src/zenml/integrations/skypilot/orchestrators/skypilot_base_vm_orchestrator.py

```diff
@@ -0,0 +1,251 @@
+            )
+
+        # Set up some variables for configuration
+        orchestrator_run_id = str(uuid4())
+        environment[
+            ENV_ZENML_SKYPILOT_ORCHESTRATOR_RUN_ID
+        ] = orchestrator_run_id
+
+        settings = cast(
+            SkypilotBaseOrchestratorSettings,
+            self.get_settings(deployment),
+        )
+
+        entrypoint = PipelineEntrypointConfiguration.get_entrypoint_command()
+        entrypoint_str = " ".join(entrypoint)
+        arguments = PipelineEntrypointConfiguration.get_entrypoint_arguments(
+            deployment_id=deployment.id
+        )
+        arguments_str = " ".join(arguments)
+
+        # Set up docker run command
+        image = self.get_image(deployment=deployment)
+        docker_environment_str = " ".join(
+            f"-e {k}={v}" for k, v in environment.items()
+        )
+
+        start_time = time.time()
+
+        instance_type = settings.instance_type or self.DEFAULT_INSTANCE_TYPE
+
+        # Set up credentials
+        self.setup_credentials()
+
+        # Run the entire pipeline
+        try:
+            task = sky.Task(
+                run=f"docker run --rm {docker_environment_str} {image} {entrypoint_str} {arguments_str}",
+                setup=self.get_setup(stack),
+            )
+            task = task.set_resources(
+                sky.Resources(
+                    cloud=self.cloud,
+                    instance_type=instance_type,
+                    cpus=settings.cpus,
+                    memory=settings.memory,
+                    accelerators=settings.accelerators,
+                    accelerator_args=settings.accelerator_args,
+                    use_spot=settings.use_spot,
+                    spot_recovery=settings.spot_recovery,
+                    region=settings.region,
+                    zone=settings.zone,
+                    image_id=settings.image_id,
+                    disk_size=settings.disk_size,
+                    disk_tier=settings.disk_tier,
+                )
+            )
+
+            cluster_name = settings.cluster_name
+            if cluster_name is None:
+                # Find existing cluster
+                for i in sky.status(refresh=True):
+                    if isinstance(
+                        i["handle"].launched_resources.cloud, type(self.cloud)
+                    ):
+                        cluster_name = i["handle"].cluster_name
+                        logger.info(
+                            f"Found existing cluster {cluster_name}. Reusing..."
+                        )
+
+            sky.launch(
+                task,
+                cluster_name,
+                retry_until_up=settings.retry_until_up,
+                idle_minutes_to_autostop=settings.idle_minutes_to_autostop,
+                down=settings.down,
+                stream_logs=settings.stream_logs,
+            )
+
+        except Exception as e:
+            raise RuntimeError(e)
```

Comment:
What's the purpose of reraising this as a different error?

Reply:
@fa9r you think this must be left to return skypilot exceptions rather than default them to runtime?

Reply:
Yes if there's no specific reason for it then I'll suggest to remove this since reraising errors makes debugging harder